### PR TITLE
[#2147] use static_cast when instantiating ios_base::openmode with an int

### PR DESCRIPTION
**Pull request by SwooshyCueb**
Addresses #2147

Still needs to be tested against libc++

### CODE REVIEW

File: s3/s3_operations.cpp [hunk 1]
```diff
@@ -431,7 +431,7 @@ namespace irods_s3 {
                 (oflag & O_ACCMODE) == O_WRONLY, (oflag & O_ACCMODE) == O_RDWR, (oflag & O_ACCMODE) == O_RDONLY,
                 (oflag & O_TRUNC) != 0, (oflag & O_CREAT) != 0, (oflag & O_APPEND) != 0);
 
-        ios_base::openmode mode = 0;
+        ios_base::openmode mode = static_cast<ios_base::openmode>(0);
```
Comment (korydraughn) [inline s3/s3_operations.cpp:434]: This is fine, but I'm just now noticing we don't guard against invalid values for `oflag` (i.e. invalid access mode).

This can also be handled in a separate PR.

**Comment by SwooshyCueb** [comment]
Works fine building against libc++ too, marking as ready for review

File: s3/s3_operations.cpp (hunk 1 above)
Comment (SwooshyCueb) [reply s3/s3_operations.cpp:434]: Should I open an issue for it?

**Review by korydraughn — APPROVED** [review]
I think this is safe.

Pound it.

**Comment by SwooshyCueb** [comment]
\#'d
